src/App.tsx

```diff
@@ -1,11 +1,44 @@
+import { useState } from "react";
+
+import { DisplayCard } from "./components/DisplayCard/DisplayCard";
+import { fetchSingleCharacter } from "./utils/fetchFunctions";
+
 const App = () => {
-	return (
-		<>
-			<div>
-				<h1>Pokémon Battle</h1>
-			</div>
-		</>
-	);
+  const [nameInputValue, setNameInputValue] = useState<string>("");
+  const [entities, setEntities] = useState<any[]>([]);
+
+  const handleSubmit = async () => {
+    setEntities([...entities, await fetchSingleCharacter(nameInputValue)]);
+    setNameInputValue("");
+  };
+
+  return (
+    <>
+      <div>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleSubmit();
+          }}
+        >
+          <input
+            value={nameInputValue}
+            type="text"
+            placeholder="Type a name..."
+            onChange={(e) => {
+              setNameInputValue(e.target.value);
+            }}
+          />
+          <button onClick={handleSubmit}>Click me</button>
```

Comment:
this button shouldn't need an `onClick` event. a button by default is of type "submit" and a button of type "submit" by default will submit the form that it is a part of.

If you check the dev tools when you are running the app, you will see that the fetch is happening twice every time you click the button. Once from the form onSubmit and once from the button onClick.